src/components/ResourceList/ResourceList.tsx

```diff
@@ -192,417 +198,144 @@ class ResourceListInner extends React.Component<CombinedProps, State> {
-                  <EventListener event="resize" handler={this.handleResize} />
-                  {headerWrapperOverlay}
-                  <div className={styles.HeaderContentWrapper}>
-                    {headerTitleMarkup}
-                    {checkableButtonMarkup}
-                    {alternateToolMarkup}
-                    {sortingSelectMarkup}
-                    {selectButtonMarkup}
-                  </div>
-                  {bulkActionsMarkup}
-                </div>
-              );
-            }}
-          </Sticky>
-        </div>
-      );
-
-    const emptyStateMarkup = showEmptyState ? (
-      <div className={styles.EmptySearchResultWrapper}>
-        <EmptySearchResult {...this.emptySearchResultText()} withIllustration />
-      </div>
-    ) : null;
-
-    const defaultTopPadding = 8;
-    const topPadding =
-      loadingPosition > 0 ? loadingPosition : defaultTopPadding;
-    const spinnerStyle = {paddingTop: `${topPadding}px`};
-
-    const spinnerSize = items.length < 2 ? 'small' : 'large';
-
-    const loadingOverlay = loading ? (
-      <React.Fragment>
-        <div className={styles.SpinnerContainer} style={spinnerStyle}>
-          <Spinner size={spinnerSize} accessibilityLabel="Items are loading" />
-        </div>
-        <div className={styles.LoadingOverlay} />
-      </React.Fragment>
-    ) : null;
-
-    const className = classNames(
-      styles.ItemWrapper,
-      loading && styles['ItemWrapper-isLoading'],
-    );
-    const loadingWithoutItemsMarkup =
-      loading && !this.itemsExist() ? (
-        <div className={className} tabIndex={-1}>
-          {loadingOverlay}
-        </div>
-      ) : null;
-
-    const resourceListClassName = classNames(
-      styles.ResourceList,
-      loading && styles.disabledPointerEvents,
-      selectMode && styles.disableTextSelection,
-    );
-
-    const listMarkup = this.itemsExist() ? (
-      <ul
-        className={resourceListClassName}
-        ref={this.listRef}
-        aria-live="polite"
-        aria-busy={loading}
-      >
-        {loadingOverlay}
-        {items.map(this.renderItem)}
-      </ul>
-    ) : (
-      emptyStateMarkup
-    );
-
-    const context = {
-      selectable: this.selectable(),
-      selectedItems,
-      selectMode,
-      resourceName,
-      loading,
-      onSelectionChange: this.handleSelectionChange,
-      registerCheckableButtons: this.handleCheckableButtonRegistration,
-    };
+  };
 
-    return (
-      <ResourceListContext.Provider value={context}>
-        <div className={styles.ResourceListWrapper}>
-          {filterControlMarkup}
-          {headerMarkup}
-          {listMarkup}
-          {loadingWithoutItemsMarkup}
-        </div>
-      </ResourceListContext.Provider>
-    );
+  function getAllItemsOnPage(
+    items: any,
+    idForItem: (item: any, index: number) => string,
+  ) {
+    return items.map((item: any, index: number) => {
+      return idForItem(item, index);
+    });
```

Comment:
We can define this outside of the component since it's not any context outside of itself

Reply:
This works! committed.